### PR TITLE
Notebook cleaned and data on S3 - Linear learner MNIST

### PR DESCRIPTION
**Pull request by vivekmadan2**
1. Data stored on S3 bucket offline and loaded from bucket in the notebook instead of downloading and uploading.
2. Sagemaker interface has changed and the notebook was broken. Fix applied.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by mabunday — CHANGES_REQUESTED** [review]

**Review by mabunday — APPROVED** [review]

**Review by mabunday — APPROVED** [review]

File: introduction_to_amazon_algorithms/linear_learner_mnist/linear_learner_mnist.ipynb [hunk 1]
```diff
@@ -245,27 +249,25 @@
...
-    "                                       sagemaker_session=sess)\n",
-    "linear.set_hyperparameters(feature_dim=784,\n",
-    "                           predictor_type='binary_classifier',\n",
-    "                           mini_batch_size=200)\n",
-    "\n",
-    "linear.fit({'train': s3_train_data})"
+    "linear = sagemaker.estimator.Estimator(\n",
+    "    container,\n",
+    "    role,\n",
+    "    train_instance_count=1,\n",
```
Comment (aaronmarkham) [inline introduction_to_amazon_algorithms/linear_learner_mnist/linear_learner_mnist.ipynb:263]: not v2 params... should be instance_count
```suggestion
    "    instance_count=1,\n",
```

File: introduction_to_amazon_algorithms/linear_learner_mnist/linear_learner_mnist.ipynb [hunk 2]
```diff
@@ -245,27 +249,25 @@
...
+    "    train_instance_type=\"ml.c4.xlarge\",\n",
```
Comment (aaronmarkham) [inline introduction_to_amazon_algorithms/linear_learner_mnist/linear_learner_mnist.ipynb:264]:
```suggestion
    "    instance_type=\"ml.c4.xlarge\",\n",
```

**Review by aaronmarkham — COMMENTED** [review]
Shouldn't this be using SM PySDK v2?

File: introduction_to_amazon_algorithms/linear_learner_mnist/linear_learner_mnist.ipynb (hunk 1 above)
Comment (vivekmadan2) [reply introduction_to_amazon_algorithms/linear_learner_mnist/linear_learner_mnist.ipynb]: Both train_instance_type and train_instance_count are supported in v2 as well. Though, as per your suggestion, I changed them to instance_type and instance_count in all the notebooks.

Let's do other such changes after re:invent. Right now, priority is functioning notebooks.

**Review by mabunday — APPROVED** [review]

File: introduction_to_amazon_algorithms/linear_learner_mnist/linear_learner_mnist.ipynb [hunk 3]
```diff
@@ -110,25 +118,25 @@
...
     "\n",
-    "def show_digit(img, caption='', subplot=None):\n",
-    "    if subplot==None:\n",
-    "        _,(subplot)=plt.subplots(1,1)\n",
-    "    imgr=img.reshape((28,28))\n",
-    "    subplot.axis('off')\n",
-    "    subplot.imshow(imgr, cmap='gray')\n",
+    "\n",
+    "def show_digit(img, caption=\"\", subplot=None):\n",
+    "    if subplot == None:\n",
```
[the remark this answers is not in the document]
Comment (vivekmadan2) [reply introduction_to_amazon_algorithms/linear_learner_mnist/linear_learner_mnist.ipynb]: Changed.

**Review by aaronmarkham — APPROVED** [review]

**Pull request merged**